CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -480,30 +480,15 @@ public static function load(&$group, $force = FALSE) {
       return;
     }
 
-    // grab a lock so other processes don't compete and do the same query
-    $lock = Civi::lockManager()->acquire("data.core.group.{$groupID}");
```

Comment:
Why are we removing this - seems like a good thing?

Reply:
@eileenmcnaughton its moved further lower

Reply:
Yes but why? It seems like a good thing where it was

Reply:
We should only lock the database for as short a time as we need and that is when we are doing the actual operations on civicrm_group_contact_cache. We're using a temp table much more than what the previous code was doing

Reply:
We're not locking the database though - we are specifically telling other processes that want to rebuild the cache for the specific group not to because a process is already doing that

Reply:
It's as @seamuslee001 says.  We're telling other processes via a static `self::$_alreadyLoaded[$groupID] = 1;` at the start of the function that we're already building. But the building can take a significant (in cpu/database terms) amount of time and we only want the mysql lock (which is what Civi::lockmanager() gives us) for the shortest amount of time which is why it's moved down.

Reply:
I still think the broken lock system should be gotten rid of, rather than allowing it to be configured to work on certain versions of mysql.  It causes weird side effects like this desire to minimize the time that a locks are held. A simple database table for scoped locks works fine.

Reply:
I don't think it IS broken if you have the define set - which @mattwire proposes to do later on.

@mattwire I think you are saying 'we only really need to lock it when we are moving data from the temp table to the live table'. So that means we have 2 processes 

Process one 
- starts potentially slow query to fill temp table
- grabs lock
- moves data from temp table to live table
- releases lock

Process two
- starts potentially slow query to fill temp table
- tries to grab lock & fails
- aborts

Doing the potentially slow query is only useful if it will then be able to do something with the results. The alternative is 

Process one 
- starts potentially-slow query to fill temp table
- grabs lock
- moves data from temp table to live table
- releases lock

Process two
- tries to grab lock & fails
- aborts without running potentially-slow query to fill temp table

That seems better to me - especially since it might not be process 2 running the potentially slow query but processes 2,3,4,5,6,7,8,9,10.